### PR TITLE
Support multiple hostnames

### PR DESCRIPTION
**Pull request by pschorf**

### CODE REVIEW

File: waiter/src/waiter/core.clj [hunk 1]
```diff
@@ -700,8 +700,12 @@
    :waiter-request?-fn (pc/fnk [[:settings hostname]]
                          (let [local-router (InetAddress/getLocalHost)
                                waiter-router-hostname (.getCanonicalHostName local-router)
-                               waiter-router-ip (.getHostAddress local-router)]
-                           (waiter-request?-factory (set [hostname waiter-router-hostname waiter-router-ip]))))
+                               waiter-router-ip (.getHostAddress local-router)
+                               hostnames (if (sequential? hostname)
+                                           (clojure.set/union (set hostname)
+                                                              #{waiter-router-hostname waiter-router-ip})
+                                           (set [hostname waiter-router-hostname waiter-router-ip]))]
```
Comment (shamsimam) [inline waiter/src/waiter/core.clj]: Nitpick: Can we use the set literal here?

File: waiter/src/waiter/core.clj [hunk 2]
```diff
@@ -700,8 +700,12 @@
    :waiter-request?-fn (pc/fnk [[:settings hostname]]
                          (let [local-router (InetAddress/getLocalHost)
                                waiter-router-hostname (.getCanonicalHostName local-router)
-                               waiter-router-ip (.getHostAddress local-router)]
-                           (waiter-request?-factory (set [hostname waiter-router-hostname waiter-router-ip]))))
+                               waiter-router-ip (.getHostAddress local-router)
+                               hostnames (if (sequential? hostname)
+                                           (clojure.set/union (set hostname)
```
Comment (shamsimam) [inline waiter/src/waiter/core.clj]: `clojure.set` is already imported, we can just do `set/union`

File: waiter/src/waiter/core.clj [hunk 3]
```diff
@@ -700,8 +700,12 @@
    :waiter-request?-fn (pc/fnk [[:settings hostname]]
                          (let [local-router (InetAddress/getLocalHost)
                                waiter-router-hostname (.getCanonicalHostName local-router)
-                               waiter-router-ip (.getHostAddress local-router)]
-                           (waiter-request?-factory (set [hostname waiter-router-hostname waiter-router-ip]))))
+                               waiter-router-ip (.getHostAddress local-router)
+                               hostnames (if (sequential? hostname)
```
Comment (shamsimam) [inline waiter/src/waiter/core.clj]: +1 on use of `sequential?`

File: waiter/test/waiter/core_test.clj [hunk 4]
```diff
@@ -1099,3 +1099,14 @@
 
       (testing "require-authentication"
         (is (= {:source :spnego-handler} (request-handler {:headers {}})))))))
+
+(deftest test-waiter-request?-fn
```
Comment (shamsimam) [inline waiter/test/waiter/core_test.clj]: The test should also include negative scenarios, e.g. `(is (not (waiter-request?-fn {:headers {"host" "waiter-host-1"}})))`  in the first testing block.

File: waiter/test/waiter/core_test.clj [hunk 5]
```diff
@@ -1099,3 +1099,14 @@
...
+  (testing "string hostname config"
+    (let [config {:settings {:hostname "waiter-host"}}
+          waiter-request?-fn ((:waiter-request?-fn routines) config)]
+      (is (waiter-request?-fn {:headers {"host" "waiter-host"}}))))
+  (testing "list hostname config"
```
Comment (shamsimam) [inline waiter/test/waiter/core_test.clj]: Nitpick: separate these two `testing` blocks with a whitepsace.

File: waiter/src/waiter/core.clj (hunk 2 above)
Comment (pschorf) [reply waiter/src/waiter/core.clj]: Done

File: waiter/src/waiter/core.clj (hunk 1 above)
Comment (pschorf) [reply waiter/src/waiter/core.clj]: Done

File: waiter/test/waiter/core_test.clj (hunk 4 above)
Comment (pschorf) [reply waiter/test/waiter/core_test.clj]: Done

File: waiter/test/waiter/core_test.clj (hunk 5 above)
Comment (pschorf) [reply waiter/test/waiter/core_test.clj]: Done

**Comment by sradack** [comment]
Why did we merge before a passing build?

File: waiter/src/waiter/core.clj [hunk 6]
```diff
@@ -700,8 +700,12 @@
    :waiter-request?-fn (pc/fnk [[:settings hostname]]
```
Comment (sradack) [inline waiter/src/waiter/core.clj]: This isn't the only place `hostname` is used.

**Pull request merged**